<a href="https://colab.research.google.com/github/juancamilo9/works_data_science/blob/master/Ejercicio_de_encontrar_la_altura_promedio_por_ol%C3%ADmpico_individual_por_a%C3%B1o_(base).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
import pandas as pd
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
filename='/content/drive/MyDrive/athleteEventsNoPersonal.csv'
df = pd.read_csv(filename)
df.head()

ID   Age  Height         Team  NOC        Games  Year  Season         City  \
0   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
1   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
2   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
3   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
4   5  27.0   185.0  Netherlands  NED  1994 Winter  1994  Winter  Lillehammer   

           Sport                               Event Medal  
0  Speed Skating    Speed Skating Women's 500 metres   NaN  
1  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
2  Speed Skating    Speed Skating Women's 500 metres   NaN  
3  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
4  Speed Skating    Speed Skating Women's 500 metres   NaN

In [10]:
# Filas y Columnas del DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


El data Frame cuenta con 12 columnas y 40616 entradas o filas.
Hay valores Faltantes en las columnas 'Age', 'Height', y 'Medal'
Pienso que la razon por la cual faltan tantos datos en la columna de 'Medal' es por que el Dataframe contiene información con respecto a los participantes de los juegos olimpicos, pero el hecho de participar en una competencia no lo hace acreedor de una medalla.

**. ¿Cuántas entradas corresponden a la ciudad de Londres?**

In [34]:
filter = df['City'] == 'London'

df[filter]


ID   Age  Height     Team  NOC        Games  Year  Season    City  \
36         54   NaN     NaN     Iran  IRI  1948 Summer  1948  Summer  London   
42         67  26.0   179.0   Russia  RUS  2012 Summer  2012  Summer  London   
66        155  34.0   185.0    Egypt  EGY  2012 Summer  2012  Summer  London   
69        159  19.0   167.0    Egypt  EGY  2012 Summer  2012  Summer  London   
72        185  31.0     NaN    Egypt  EGY  1948 Summer  1948  Summer  London   
...       ...   ...     ...      ...  ...          ...   ...     ...     ...   
40577  135489  22.0   182.0   Russia  RUS  2012 Summer  2012  Summer  London   
40578  135489  22.0   182.0   Russia  RUS  2012 Summer  2012  Summer  London   
40590  135517  22.0     NaN  Austria  AUT  1948 Summer  1948  Summer  London   
40595  135528  28.0   181.0  Germany  GER  2012 Summer  2012  Summer  London   
40609  135558  32.0   200.0   Poland  POL  2012 Summer  2012  Summer  London   

                       Sport                                         Event  \
36                    Boxing                      Boxing Men's Heavyweight   
42                 Athletics               Athletics Women's Javelin Throw   
66                 Wrestling      Wrestling Men's Heavyweight, Greco-Roman   
69     Synchronized Swimming            Synchronized Swimming Women's Team   
72                   Fencing                Fencing Men's Foil, Individual   
...                      ...                                           ...   
40577               Swimming        Swimming Women's 200 metres Backstroke   
40578               Swimming  Swimming Women's 4 x 100 metres Medley Relay   
40590               Swimming          Swimming Men's 100 metres Backstroke   
40595              Badminton                       Badminton Men's Singles   
40609             Volleyball                   Volleyball Men's Volleyball   

        Medal  
36        NaN  
42        NaN  
66        NaN  
69        NaN  
72        NaN  
...       ...  
40577  Silver  
40578     NaN  
40590     NaN  
40595     NaN  
40609     NaN  

[3370 rows x 12 columns]

3370 entradas, corresponden a 'London'

# ¿Cuántos años tiene el atleta más joven en nuestros datos de muestra?  Pista: usen min()

In [35]:
df['Age'].min()
# ¡11 Años¡¡

11.0

#  Creen filtros para encontrar:

**¿Cuántos atletas que participaron en el deporte "Tug-Of-War" (tira y afloja) pertenecían en el equipo “Sweden” (sueco) en el conjunto de datos de muestra?**

In [71]:
# filto prar 'Tug-Of-War'
filter_TW = df['Sport'] == 'Tug-Of-War'
# filtro para 'Sweden'
filter_sweden = df['Team'] == 'Sweden'
# cuento con respecto al ID del atleta
print("Numero Total de entradas:" , df[filter_TW & filter_sweden]['ID'].count())
df[filter_TW & filter_sweden]



Numero Total de entradas: 6


ID   Age  Height    Team  NOC        Games  Year  Season    City  \
10023  34205  22.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
12681  43105  23.0     NaN  Sweden  SWE  1906 Summer  1906  Summer  Athina   
13015  44087  29.0   182.0  Sweden  SWE  1906 Summer  1906  Summer  Athina   
16250  55169  23.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
19048  64266  21.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
20550  68647  26.0   190.0  Sweden  SWE  1906 Summer  1906  Summer  Athina   

            Sport                        Event   Medal  
10023  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
12681  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze  
13015  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze  
16250  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
19048  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
20550  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze

# b. ¿Cuántas medallas de oro se le entregaron a los atletas que tenían más de 40 años?

In [81]:
filter_age = df['Age'] > 40
filter_medal = df['Medal'] == 'Gold'
# contamos con respecto al ID del atleta
print("Total de atletas mayores de 40 años que ganaron medalla de oro:" , df[filter_medal & filter_age]['ID'].count())
df[filter_medal & filter_age]

Total de atletas mayores de 40 años que ganaron medalla de oro: 66


ID   Age  Height                Team  NOC        Games  Year  \
740      2735  41.0   168.0              Russia  RUS  2000 Summer  2000   
1103     4029  46.0     NaN           Argentina  ARG  1936 Summer  1936   
1421     4993  41.0   180.0             Mosk II  NOR  1920 Summer  1920   
3040    10769  49.0   180.0          White Lady  DEN  1964 Summer  1964   
3270    11617  44.0     NaN             Heroine  GBR  1908 Summer  1908   
...       ...   ...     ...                 ...  ...          ...   ...   
38916  130271  43.0     NaN            Heira II  NOR  1920 Summer  1920   
38952  130381  42.0   167.0       United States  USA  1972 Summer  1972   
39193  130936  56.0     NaN       United States  USA  1908 Summer  1908   
39197  130936  60.0     NaN       United States  USA  1912 Summer  1912   
39365  131533  58.0     NaN  Cincinnati Archers  USA  1904 Summer  1904   

       Season       City             Sport  \
740    Summer     Sydney          Shooting   
1103   Summer     Berlin              Polo   
1421   Summer  Antwerpen           Sailing   
3040   Summer      Tokyo           Sailing   
3270   Summer     London           Sailing   
...       ...        ...               ...   
38916  Summer  Antwerpen           Sailing   
38952  Summer     Munich           Archery   
39193  Summer     London          Shooting   
39197  Summer  Stockholm  Art Competitions   
39365  Summer  St. Louis           Archery   

                                             Event Medal  
740    Shooting Men's Rapid-Fire Pistol, 25 metres  Gold  
1103                               Polo Men's Polo  Gold  
1421                       Sailing Mixed 10 metres  Gold  
3040           Sailing Mixed Three Person Keelboat  Gold  
3270                        Sailing Mixed 7 metres  Gold  
...                                            ...   ...  
38916                      Sailing Mixed 12 metres  Gold  
38952                   Archery Women's Individual  Gold  
39193   Shooting Men's Running Target, Double Shot  Gold  
39197           Art Competitions Mixed Sculpturing  Gold  
39365                   Archery Women's Team Round  Gold  

[66 rows x 12 columns]

# Determinen la altura promedio de:
Todas las filas incluidas.

In [82]:
df['Height'].mean()

175.3811165585956

***Todas las filas incluidas, pero agrupadas por Event (evento).***

In [87]:
df_mean = df.groupby('Event')['Height'].mean()


Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64